### PR TITLE
Add database query summary metrics

### PR DESCRIPTION
**Pull request by Alistar84**
Hello!

* Type: new feature
* Link to issue: N/A

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/debugbar/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote tests for this PR
- [x] I have updated the relevant CHANGELOG
- [x] Documentation for this change is included in this PR

Small description of change:

Adds a structured summary to the Database panel with:

- total query count;
- duplicate runs, defined as executions of a normalized statement after its first execution;
- accumulated SQL execution time.

For example, executing the same normalized statement three times contributes two duplicate runs. Statement normalization collapses whitespace and deliberately ignores binding values, so repeated executions of one prepared statement are grouped together.

Each repeated row exposes its total `occurrences` as structured data. The frontend uses that value to apply a duplicate-row CSS class and display `Executed N times` without changing the duration label. The summary remains visible with zero values when no queries are executed.

The collector and renderer tests cover repeated statements with different bindings, unique statements, per-row occurrence counts, summary values, and the zero-query state. The behavior is also documented in `docs/index.md`.

Thanks

### CODE REVIEW

File: src/DebugBar/Collector/DatabaseCollector.php [hunk 1]
```diff
@@ -63,21 +66,44 @@ final class DatabaseCollector extends AbstractCollector implements Subscriber
...
+        $duplicates = 0;
+        foreach ($occurrences as $occurrenceCount) {
+            $duplicates += $occurrenceCount - 1;
+        }
+
+        foreach ($this->queries as $query) {
+            $occurrenceCount = $occurrences[$this->queryKey($query['sql'])];
+            $label           = $this->nanosToMs($query['time']);
+            if ($occurrenceCount > 1) {
+                $label .= ' - duplicate x' . $occurrenceCount;
```
Comment (niden) [inline src/DebugBar/Collector/DatabaseCollector.php]: This one prefixes the label so that is the only way we have the indication of a duplicate in the collection. I would prefer some sort of a row flag in the array and an accompanying css class to distinguish those. This kind of contradicts the description

File: tests/Unit/DebugBar/Collector/DatabaseCollectorTest.php [hunk 2]
```diff
@@ -97,7 +120,6 @@ public function testLabelReflectsQueryDurationInMilliseconds(): void
         $this->assertSame('SELECT 1', $row['message']);
         $this->assertSame(1, $envelope['badge']);
     }
-
```
Comment (niden) [inline tests/Unit/DebugBar/Collector/DatabaseCollectorTest.php]: I presume this was removed by accident?

File: src/DebugBar/Collector/DatabaseCollector.php [hunk 3]
```diff
@@ -63,21 +66,44 @@ final class DatabaseCollector extends AbstractCollector implements Subscriber
...
-
         return [
-            'panel' => $rows,
-            'badge' => count($this->queries),
+            'panel'   => $rows,
+            'badge'   => count($this->queries),
+            'summary' => [
+                'Queries'     => count($this->queries),
+                'Duplicates' => $duplicates,
+                'Total time'  => $this->nanosToMs($totalNanoseconds),
```
Comment (niden) [inline src/DebugBar/Collector/DatabaseCollector.php]: The keys here are labels that are printed on screen. This is fine but we are locking ourselves in a format that might not help in the future. Change them to their lowercase and snake case equivalent, so that `titleize()` in the FE can make them readable. The stan shapes will become more coherent after this.

Comment (niden) [reply src/DebugBar/Collector/DatabaseCollector.php]: Another option is to follow what the list panel does with the label/value pairs, being represented here as a list of label/value elements. Then in the JS you can just iterate them. That would allow for expansion in the future and not bind into a specific format that any future implementation will have to follow. Your choice.

File: src/DebugBar/Collector/DatabaseCollector.php [hunk 4]
```diff
@@ -63,21 +66,44 @@ final class DatabaseCollector extends AbstractCollector implements Subscriber
...
+        foreach ($occurrences as $occurrenceCount) {
+            $duplicates += $occurrenceCount - 1;
+        }
+
+        foreach ($this->queries as $query) {
+            $occurrenceCount = $occurrences[$this->queryKey($query['sql'])];
+            $label           = $this->nanosToMs($query['time']);
+            if ($occurrenceCount > 1) {
+                $label .= ' - duplicate x' . $occurrenceCount;
+            }
```
Comment (niden) [inline src/DebugBar/Collector/DatabaseCollector.php]: Also this one appends text to the label. This makes it much wider than any other ones which will have a negative effect on the display (i.e. `blah - duplicate xY`). Make this text a row attribute and leave the label as is. Something like `Executed X times` or something similar. You can pass the amount of duplicate queries per row as a number or `null` to the FE that way.

Comment (niden) [reply src/DebugBar/Collector/DatabaseCollector.php]: I also considered the text to be reduced i.e. instead of 

`blah - duplicate x15`

to

`blah - x15`

which will certainly help but I am leaning towards the first approach with the row attribute. Thoughts?

File: src/DebugBar/Collector/DatabaseCollector.php [hunk 5]
```diff
@@ -63,21 +66,44 @@ final class DatabaseCollector extends AbstractCollector implements Subscriber
...
+     * @return database_envelope
      */
     public function collect(): array
     {
-        $rows = [];
+        $rows              = [];
+        $occurrences       = [];
+        $totalNanoseconds  = 0;
         foreach ($this->queries as $query) {
+            $key               = $this->queryKey($query['sql']);
```
Comment (niden) [inline src/DebugBar/Collector/DatabaseCollector.php]: This gets called twice one on this loop and one in the one below. Add a map array on this loop and use that on the next. Small cycles yes I agree and performance is negligible, but in the future one change here will not be propagated below.

File: src/DebugBar/Collector/DatabaseCollector.php [hunk 6]
```diff
@@ -63,21 +66,44 @@ final class DatabaseCollector extends AbstractCollector implements Subscriber
...
                 'message' => $this->formatQuery($query['sql'], $query['bindings']),
             ];
         }
-
         return [
-            'panel' => $rows,
-            'badge' => count($this->queries),
+            'panel'   => $rows,
+            'badge'   => count($this->queries),
+            'summary' => [
```
Comment (niden) [inline src/DebugBar/Collector/DatabaseCollector.php]: I have not run this locally but this change increases the data sent to the FE by including the summary always. What would happen if no queries run and this element is empty? Do we just show zero counts? This is a behavior question, not asking a change here if you don't think it is warranted. This behavior here (whatever it will end up being) needs to make its way in the docs. _if no queries are executed the numbers stay 0_ or something like that.

File: tests/Unit/DebugBar/Collector/DatabaseCollectorTest.php [hunk 7]
```diff
@@ -46,6 +46,23 @@ public function testBindingsAreAppendedToTheQuery(): void
         );
     }
 
+    public function testDuplicateDetectionIgnoresBindingsButNotDifferentStatements(): void
+    {
+        $collector = new DatabaseCollector();
+        $property = new ReflectionProperty(DatabaseCollector::class, 'queries');
+        $property->setValue($collector, [
```
Comment (niden) [inline tests/Unit/DebugBar/Collector/DatabaseCollectorTest.php]: Can you expand this to have more statements? This way it is not very similar with the Duration one below. Add a few more statements here to ensure that we are testing the executions counting as duplicates, vs unique ones count as unique.

**Review by niden — CHANGES_REQUESTED** [review]
Thank you for this work. Some comments that I would like see addressed - some nitpicks but some design decisions. 

Thanks again, this looks super!

File: src/DebugBar/Collector/DatabaseCollector.php (hunk 1 above)
Comment (Alistar84) [reply src/DebugBar/Collector/DatabaseCollector.php]: Addressed in 2a2bcd6. Each database row now exposes an `occurrences` value (`int|null`), and the frontend derives the `is-duplicate` CSS class from it.

File: tests/Unit/DebugBar/Collector/DatabaseCollectorTest.php (hunk 2 above)
Comment (Alistar84) [reply tests/Unit/DebugBar/Collector/DatabaseCollectorTest.php]: Yes, that blank line was removed accidentally. It has been restored in 2a2bcd6.

File: src/DebugBar/Collector/DatabaseCollector.php (hunk 3 above)
Comment (Alistar84) [reply src/DebugBar/Collector/DatabaseCollector.php]: Addressed. Summary identifiers are now lowercase snake_case (`queries`, `duplicates`, and `total_time`), and the frontend renders them through `titleize()`.

Comment (Alistar84) [reply src/DebugBar/Collector/DatabaseCollector.php]: I chose the extensible list representation suggested here. A summary is now a list of `label`/`value` elements, so future fields can be added without binding the renderer to object keys.

File: src/DebugBar/Collector/DatabaseCollector.php (hunk 4 above)
Comment (Alistar84) [reply src/DebugBar/Collector/DatabaseCollector.php]: Addressed. The duration label is left unchanged. Repeated statements carry an `occurrences` row attribute, and the frontend displays `Executed N times` separately.

Comment (Alistar84) [reply src/DebugBar/Collector/DatabaseCollector.php]: Agreed - I implemented the row-attribute approach rather than adding duplicate text to the duration label.

File: src/DebugBar/Collector/DatabaseCollector.php (hunk 5 above)
Comment (Alistar84) [reply src/DebugBar/Collector/DatabaseCollector.php]: Addressed. Normalized SQL keys are collected once in a `$queryKeys` map and reused while building the rows. Duplicate totals are also accumulated in the first pass.

File: src/DebugBar/Collector/DatabaseCollector.php (hunk 6 above)
Comment (Alistar84) [reply src/DebugBar/Collector/DatabaseCollector.php]: The summary intentionally remains visible with `0` queries, `0` duplicates, and `0ms`. This behavior is now explicitly documented and covered by a test.

File: tests/Unit/DebugBar/Collector/DatabaseCollectorTest.php (hunk 7 above)
Comment (Alistar84) [reply tests/Unit/DebugBar/Collector/DatabaseCollectorTest.php]: Expanded in 2a2bcd6. The test now uses five statements: three executions of the same prepared statement with different bindings and two unique statements. It asserts both the duplicate total and per-row occurrence counts.

**Comment by Alistar84** [comment]
Thanks for the review. I addressed all the requested changes in 2a2bcd6:

- duplicate information is now exposed through a structured `occurrences` row attribute;
- query durations are no longer modified;
- duplicate rows receive a dedicated CSS class and display `Executed N times`;
- summaries now use an extensible list of `label`/`value` elements with snake_case identifiers;
- normalized query keys are computed once and reused;
- zero-query behavior is documented;
- duplicate detection tests now cover multiple repeated and unique statements.

The relevant tests and static analysis checks pass locally. Could you please take another look?

**Comment by niden** [comment]
@Alistar84 Thank you for the quick turnaround. One note before I go for second pass, reminder https://docs.phalcon.io/5.18/ai-development/. If you need to change the commit messages go for it.

**Comment by niden** [comment]
@Alistar84 Also before I forget, this will need a few lines in docs/index because it is new behavior

File: src/DebugBar/Collector/DatabaseCollector.php [hunk 8]
```diff
@@ -63,21 +66,42 @@ final class DatabaseCollector extends AbstractCollector implements Subscriber
...
+     * @return database_envelope
      */
     public function collect(): array
     {
-        $rows = [];
+        $rows              = [];
+        $occurrences       = [];
+        $queryKeys         = [];
+        $duplicates        = 0;
+        $totalNanoseconds  = 0;
```
Comment (niden) [inline src/DebugBar/Collector/DatabaseCollector.php]: We can do one more optimization here. Have a `$keys` array and store the `qyeryKey` there. Then in the second loop you use the `$keys` array. 

Example in the first loop

```php
foreach ($this->queries as $index => $key) {
   $key = $this->queryKey($query['sql']);
   $keys[$index] = $key;
...
}

foreach ($this->queries as $index => $key) {
     $key = $keys[$index];
...
}
```

Saves on the queryKey calls.

File: src/DebugBar/Collector/DatabaseCollector.php [hunk 9]
```diff
@@ -63,21 +66,42 @@ final class DatabaseCollector extends AbstractCollector implements Subscriber
...
         }
 
         return [
-            'panel' => $rows,
-            'badge' => count($this->queries),
+            'panel'   => $rows,
+            'badge'   => count($this->queries),
+            'summary' => [
+                ['label' => 'queries', 'value' => count($this->queries)],
+                ['label' => 'duplicates', 'value' => $duplicates],
```
Comment (niden) [inline src/DebugBar/Collector/DatabaseCollector.php]: Let us change this key to 'duplicate_runs', because it actually the count of extra executions that are recorded

**Review by niden — CHANGES_REQUESTED** [review]
A few more issues to resolve.

Also note the AI doc in my other document (main thread) and please make sure to update the PR description as well since it is a bit off after these changes.

Thanks

File: src/DebugBar/Collector/DatabaseCollector.php (hunk 8 above)
Comment (Alistar84) [reply src/DebugBar/Collector/DatabaseCollector.php]: Thanks. The previous implementation already called `queryKey()` only once per query in the first loop and reused the value from `$queryKeys` in the second loop. Since `$this->queries` is a list, appending with `$queryKeys[]` preserved the same indexes. I have nevertheless changed it to `$queryKeys[$index] = $key` in d0a78b7 to make the index mapping explicit and align the implementation with your suggestion.

File: src/DebugBar/Collector/DatabaseCollector.php (hunk 9 above)
Comment (Alistar84) [reply src/DebugBar/Collector/DatabaseCollector.php]: Addressed in d0a78b7. The summary identifier is now `duplicate_runs`, the internal variable is `$duplicateRuns`, and the tests and documentation use the same terminology.

**Comment by Alistar84** [comment]
Thanks. The latest feedback has been addressed in d0a78b7:

- query-key caching is now expressed as an explicit index map;
- the summary metric is named `duplicate_runs` and documented as executions after the first;
- `docs/index.md`, the changelog, and the PR description now describe the current behavior;
- all three commits now carry the required `Assisted-by: Codex` trailer.

The focused collector/renderer tests, PHPStan, code style, and collector coverage checks pass locally.

File: resources/assets/debugbar.js [hunk 10]
```diff
@@ -95,8 +95,23 @@
...
+            var occurrences = Number(row.occurrences);
+
+            if (occurrences > 1) {
+                tr.classList.add('is-duplicate');
+            }
+
             tr.appendChild(el('td', 'phalcon-debugbar-key', scalar(row.label)));
-            tr.appendChild(el('td', 'phalcon-debugbar-value', scalar(row.message)));
+            value.appendChild(el('span', 'phalcon-debugbar-message', scalar(row.message)));
+            if (occurrences > 1) {
```
Comment (niden) [inline resources/assets/debugbar.js]: There are two checks for occurences > 1. Could we make this one?

File: resources/assets/debugbar.css [hunk 11]
```diff
@@ -145,6 +173,22 @@
...
+}
+
+#phalcon-debugbar .phalcon-debugbar-duplicate-count {
+    display: inline-block;
+    margin-left: 10px;
+    padding: 1px 6px;
+    border-radius: 999px;
+    background: #8a4b08;
+    color: #fff1d6;
+    font-size: 10px;
```
Comment (niden) [inline resources/assets/debugbar.css]: Nitpick but... Can we make this 11px since that is what we use everywhere else

**Review by niden — COMMENTED** [review]
@Alistar84  Two more nitpics. This is nearly ready for merge.

Thanks again

File: resources/assets/debugbar.js (hunk 10 above)
Comment (Alistar84) [reply resources/assets/debugbar.js]: Addressed in d1e6b99. The duplicate row class and occurrence badge are now handled by the same `occurrences > 1` condition.

File: resources/assets/debugbar.css (hunk 11 above)
Comment (Alistar84) [reply resources/assets/debugbar.css]: Addressed in d1e6b99. The duplicate-count font size is now `11px`, matching the surrounding debugbar styles.

**Review by niden — APPROVED** [review]

**Comment by niden** [comment]
@Alistar84 Excellent work. Thank you for this addition!
